File: java-spring/queryside-backend/src/main/java/net/chrisrichardson/eventstore/javaexamples/banking/backend/queryside/accounts/AccountInfo.java
package net.chrisrichardson.eventstore.javaexamples.banking.backend.queryside.accounts;

import java.util.List;

/**
 * Created by cer on 11/21/14.
 */
public class AccountInfo {

  private String id;
  private long balance;
  private List<AccountChangeInfo> changes;
  private List<AccountTransactionInfo> transactions;
  private String version;

  private AccountInfo() {
  }

  public AccountInfo(String id, long balance, List<AccountChangeInfo> changes, List<AccountTransactionInfo> transactions, String version) {

    this.id = id;
    this.balance = balance;
    this.changes = changes;
    this.transactions = transactions;
    this.version = version;
  }

  public String getId() {
    return id;
  }

  public long getBalance() {
    return balance;
  }

  public List<AccountChangeInfo> getChanges() {
    return changes;
  }

  public List<AccountTransactionInfo> getTransactions() {
    return transactions;
  }

  public String getVersion() {
    return version;
  }
}

File: scala-spring/accounts-query-side-backend/src/main/scala/net/chrisrichardson/eventstore/examples/bank/queryside/AccountInfo.scala
package net.chrisrichardson.eventstore.examples.bank.queryside

import org.springframework.data.mongodb.repository.MongoRepository

case class AccountInfo(id : String, balance : Long,
                       changes : java.util.List[AccountChangeInfo],
                       transactions : java.util.List[AccountTransactionInfo],
                       version : String)

case class AccountChangeInfo(changeId : String, transactionId : String, transactionType : String, amount : Long, balanceDelta: Long)

case class AccountTransactionInfo(transactionId : String, fromAccountId: String, toAccountId: String, amount : Long)

trait AccountInfoRepository extends MongoRepository[AccountInfo, String]